### DISCUSSION TITLE
How to change to phong shader set?

### DISCUSSION THREAD

**Discussion by avilkinnick** [vsg-dev/VulkanSceneGraph#1310] [Q&A]
I created phong shader set and PBR shader set and added them to options.shaderSets, both of them use shader hint "VSG_ALPHA_TEST" and everythink working, but slowly, if I am trying to remove phong shader set, everything is the same, but if I remove PBR, alpha channel becomes black, so I think it uses PBR by default. How to change it to phong?

**Comment by robertosfield** [comment]
You don't provide any context for how you are creating models so I can't provide any specific answers.

As a general info, different models that are loaded vsgXchange::assimp loader can use either PBR or Phong ShaderSets depending upon the structure/settings of the model being loaded.  If the model has been created for a PBR pipeline then the PBR ShaderSet will be used, but of it's been created for a Phong pipeline then the Phong ShaderSet should be selected.

There isn't presently a scheme for forcing PBR intent models to use Phong ShaderSet and visa-versa.

**Comment by AnyOldName3** [comment]
I don't know if it's made it to a release yet, but there was a bug in vsgXchange's AssImp-based loader where some models would be assigned the PBR shaderset even when they should have used the Phong one. If you use a newer commit of that repo, you might find the problem goes away.
